### DISCUSSION TITLE
Rule request: readonly-all-the-things

### DISCUSSION THREAD

**Discussion by marklagendijk** [angular-eslint/angular-eslint#2298] [New Rule Proposals]
The new `prefer-signals` rule is awesome, but as a developer who embraced the new signals, I want to replace all non-readonly properties with `WritableSignal`.

Therefore I want a rule that enforces all class properties to be readonly, because, either:
- The value is assigned at construction, and the property can be safely converted to `readonly`.
- The property is updated after construction, so it should be replaced by a `WritatableSignal`.

Before:
```ts
class MyComponent {
  title = "test";
  text = "Hello";

  update(){
    this.text = "Hello world";
  }
}
```

After:
```ts
class MyComponent {
  readonly title = "test";
  readonly text = signal("Hello");

  update(){
    this.text.set("Hello world");
  }
}
```

I don't know what the name of this rule should be. Or whether it should be combined with `prefer-signals`.
This new rule would help immensely with making the code ready for [zoneless](https://angular.dev/guide/experimental/zoneless).

**Comment by robertIsaac** [comment]
I agree with this proposal
But private properties should not be included as error since they don't effect the change detection (since they aren't accessible in the template)

A plus is to convert properties that aren't signal and not used in the template to private as auto fix
